akka-remote/src/main/scala/akka/remote/artery/SendQueue.scala

```diff
@@ -42,26 +43,37 @@ private[remote] final class SendQueue[T] extends GraphStageWithMaterializedValue
 
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, QueueValue[T]) = {
     @volatile var needWakeup = false
-    @volatile var queue: Queue[T] = null
+    @volatile var producerQueue: Queue[T] = null
```

Comment:
Why not store it in the QueueValue instance instead? And make it a val there? Then there will be no volatile dereference on the send path either (except of course inside the queue itself).